Automattic/wp-calypso
# Purchases: Use client-side formatting for renewal price on purchases pages

#### Proposed Changes

This PR modifies the purchases pages to use client-side currency formatting for their renewal prices instead of using the formatting provided by the server. This mirrors the changes made to checkout in https://github.com/Automattic/wp-calypso/pull/71722.

Since https://github.com/Automattic/wp-calypso/pull/71387, client side price formatting is localized and often more accurate than server-side formatting. Beyond accuracy, changing all formatting to be client-side will improve consistency since there are subtle differences between the two.

This PR should not change anything significant from a user's perspective apart from making localized prices slightly more accurate for the user's locale. (Note that it resolves https://github.com/Automattic/wp-calypso/issues/50064 by using locale-aware formatting instead of relying on the `abbr` tag added by https://github.com/Automattic/wp-calypso/pull/63254.)

To do:

- [x] Update PurchaseMetaPrice. Requires D97268-code
- [x] Update the big main price. Requires https://github.com/Automattic/wp-calypso/pull/72103 and D97932-code
- [x] Update any subtext prices on the page (like for [introductory offers](https://github.com/Automattic/wp-calypso/blob/a6d3ff557dca4786f5d00ee8cbb47949b3bd3fd5/client/me/purchases/manage-purchase/purchase-meta-introductory-offer-detail.tsx#L31)). Requires D98063-code
- [x] Update prices on the purchases list page like [this](https://github.com/Automattic/wp-calypso/blob/5ed92de1dd9561f117e61a56b5f70d0e3f2aa2a1/client/me/purchases/purchase-item/index.jsx#L156).
- [x] Update prices on memberships subscription pages like [this](https://github.com/Automattic/wp-calypso/blob/213b6bb50bb686713cd2913122c21270dca982b0/client/me/purchases/membership-item/index.tsx#L23) and [here](https://github.com/Automattic/wp-calypso/blob/213b6bb50bb686713cd2913122c21270dca982b0/client/me/memberships/subscription.jsx#L105). (It looks like this is already using `formatCurrency` so nothing needs to change.)

Before, with Canadian French and USD:

<img width="541" alt="Screenshot 2023-01-13 at 2 29 37 PM" src="https://user-images.githubusercontent.com/2036909/212404292-4d65d08d-c686-4501-801f-a72fb9114950.png">

After, with Canadian French and USD:

<img width="539" alt="Screenshot 2023-01-13 at 2 35 33 PM" src="https://user-images.githubusercontent.com/2036909/212404319-faeba391-b0f0-4a3e-8ccb-fb54e1c3fce2.png">

Fixes https://github.com/Automattic/wp-calypso/issues/71723
Fixes https://github.com/Automattic/wp-calypso/issues/50064

#### Testing Instructions

- Use an account which has at least one purchased product (although ideally we should test this with many types of products: one-time purchases, plans, Jetpack purchases, introductory offers, memberships, etc.).
- Visit `/me/purchases` and click on various purchases in the list.
- Verify that the prices on the subscription pages are more-or-less identical before and after this PR.
